oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2AuthorizationRequest.java

```diff
@@ -45,7 +45,7 @@
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-4.1.1">Section 4.1.1 Authorization Code Grant Request</a>
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-4.2.1">Section 4.2.1 Implicit Grant Request</a>
  */
-public final class OAuth2AuthorizationRequest implements Serializable {
+public class OAuth2AuthorizationRequest implements Serializable {
```

Comment:
`OAuth2AuthorizationRequest` does not require any changes. One of the primary use cases for the `OAuth2AuthorizationRequestResolver` is to customize the **Authorization Request** by adding additional parameters. This will be needed for OpenID Connect flows and also can be leveraged for PKCE. Please see the [reference doc](https://docs.spring.io/spring-security/site/docs/5.1.3.RELEASE/reference/htmlsingle/#oauth2Client-authorization-request-resolver) on how to do this.

For the changes required in this PR, I would put the logic in `DefaultOAuth2AuthorizationRequestResolver` and place the PKCE parameters in `OAuth2AuthorizationRequest.additionalParameters`